CMSgov/mint-app
# [EASI-3539] New job code support for Non CMS Users

# EASI-3539

## Changes and Description

- Added support for `MINT_NON_CMS_NONPROD` and `MINT_NON_CMS` job codes
  - Users with this code should behave exactly like `MINT_USER`, except they cannot see restricted documents.
- Added support for downgrading in LaunchDarkly



## How to test this change

- Seed the DB
- Login as `ABCD` **without** the `MINT_NON_CMS_NONPROD` job code
- Navigate to `Plan with Documents`
  - Ensure you **CAN** see all 3 docs (2 unrestricted, 1 restricted)
- Log out, log back in as `ABCD` **with** the `MINT_NON_CMS_NONPROD` job code
- Navigate to `Plan with Documents`
  - Ensure you **CAN NOT** see all 3 docs (you should only see the 2 unrestricted docs) 

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
